### PR TITLE
Add number entities for battery soc reserve

### PR DESCRIPTION
**Pull request by reidjr**
Changes to number.py only. I have never contributed through github before, so not sure about protocol for changes to pull requests, so created new one.

### CODE REVIEW

File: custom_components/givenergy_local/number.py [hunk 1]
```diff
@@ -102,6 +103,46 @@ def enable_charge_target(client: GivEnergyClient) -> None:
...
+                native_unit_of_measurement=PERCENTAGE,
+            ),
+        )
+
+        # Values correspond to SOC percentage
+        self._attr_native_min_value = 4
+        self._attr_native_max_value = 100
+
+        # A 5% step size makes the slider a bit nicer to use
+        # self._attr_native_step = 5
```
Comment (cdpuk) [inline custom_components/givenergy_local/number.py:131]: Commented out code - uncommenting this line would be the simplest approach, and it keeps the sliders working in a consistent way. Note that it's purely for slider usability; a service call can set any value if required.

**Review by cdpuk — CHANGES_REQUESTED** [review]
Just the commented out code to deal with.

I'm going to ignore the failing test - there's been an update to the test framework that I'm looking at separately.

**Comment by reidjr** [comment]
Done and Done. I was worried about 5% steps starting at 4% .....

File: custom_components/givenergy_local/number.py (hunk 1 above)
Comment (reidjr) [reply custom_components/givenergy_local/number.py]: Done

**Pull request merged**